api/config/esgateway/config_request.go

```diff
@@ -89,6 +90,20 @@ func (c *ConfigRequest) Validate() error {
 // to start the service.
 func (c *ConfigRequest) PrepareSystemConfig(creds *ac.TLSCredentials) (ac.PreparedSystemConfig, error) {
 	c.V1.Sys.Tls = creds
+
+	enableSystemNameServer := c.V1.Sys.Ngx.Main.Resolvers.EnableSystemNameservers.GetValue()
```

Comment:
For safety, we try to use the accessor method `GetV1().GetSys()...` for all reads.